projects/discussion-ui/src/lib/components/discussion-details/discussion-details.component.ts

```diff
@@ -258,12 +268,54 @@ export class DiscussionDetailsComponent implements OnInit, OnDestroy {
   }
 
   logTelemetry(event, data?) {
-    const pid = _.get(data, 'pid') || _.get(data, 'mainPid') ? 
+    const pid = _.get(data, 'pid') || _.get(data, 'mainPid') ?
     {id: _.get(data, 'pid') || _.get(data, 'mainPid'), type: 'Post'} : {};
     this.telemetryUtils.uppendContext(pid);
     this.telemetryUtils.logInteract(event, NSDiscussData.IPageName.DETAILS);
   }
 
+  onEditMode(UpdatePostStatus: boolean) {
+  if (UpdatePostStatus) {
+    this.editMode = true;
+  } else {
+    this.editMode = false;
+  }
+  }
+
+  getRealtimePost(postContent: any, index: any) {
+    this.editMode = true;
+    this.editContentIndex = index;
+    this.contentPost = postContent.replace(/<[^>]*>/g, '');
+  }
+
+  updatePost(updatedPostContent: any, pid: number) {
+    this.editMode = false;
+    const req = {
+      content: updatedPostContent,
+      title: '',
+      tags : [],
+      };
+    this.discussionService.editPost(pid, req).subscribe((data: any) => {
+      window.location.reload();
+      console.log('success data', data);
+    }, (error) => {
+      console.log('e', error);
+    });
+    console.log(pid);
+  }
+
+  deletePost(postId: number) {
+    const req = {
+        uid: this.mainUid,
+        pid: postId
+    };
+    this.discussionService.deletePost(postId, req).subscribe((data: any) => {
+      window.location.reload();
```

Comment:
Why are you doing window reload? 
you just need to remove the item from the posts list(this.data) object.

In other places, the below methos is called. You can also call the same
 this.refreshPostData(this.currentActivePage);